src/gt4py/cartesian/config.py

```diff
@@ -32,6 +32,8 @@
 
 CUDA_HOST_CXX: Optional[str] = os.environ.get("CUDA_HOST_CXX", None)
 
+GT4PY_USE_HIP: int = int(os.environ.get("GT4PY_USE_HIP", 0))
```

Comment:
There should be a way to autodetect if cupy has been compiled with HIP. This is a naive suggestion to do it using dlpack, but probably there is a simpler way using some esoteric cupy API endpoint.
```suggestion
if "GT4PY_USE_HIP" in os.environ:
    GT4PY_USE_HIP: bool = bool(int(os.environ["GT4PY_USE_HIP"])) 
else:
    # Autodetect cupy with ROCm/HIP using DLPack device type
    # Reference: https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.__dlpack_device__.html#array_api.array.__dlpack_device__
    try:
        import cupy as cp
        GT4PY_USE_HIP = cp.empty((1,)).__dlpack_device__()[0] == 10
    except Exception:
        GT4PY_USE_HIP = False
```

Reply:
I partially followed your suggestion using `cupy.cuda.get_hipcc_path()`.